Assets/Scripts/Menus/PauseMenu.cs

```diff
@@ -79,13 +79,13 @@ private void ResumeInMenu()
     public void RestartInMenu()
     {
         GameManager.Restart();
-        GameManager.Resume();
     }
 
     public void LoadMainMenu()
     {
         if (!string.IsNullOrEmpty(_nameMainMenuScene))
         {
+            GameManager.Resume();
```

Comment:
Get rid of static methods in GameManager (make them non static) and inject GameManager into all other objects that depend on it. I would also rename GameManager to GameState.